pkg/util/maps/sync.go

```diff
@@ -0,0 +1,62 @@
+package maps
+
+import "sync"
+
+// Sync is a simple wrapper around sync.Map that provides type-safe methods
+type Sync[K, V any] struct {
```

Comment:
You had enough of casting it seems ;)

Reply:
Unfortunately there is nothing like that even in `go/x/exp` 😞